provider/overrides/resources.rb

```diff
@@ -19,6 +19,14 @@ module Overrides
     # This class allows them to get access to
     # Hash functions + lets the YAML parser import them.
     class OverrideResource < Google::YamlValidator
+      # This is the list of additional attributes that a provider
+      # can add through overrides
+      def self.attributes
+        []
```

Comment:
Yes it should. Because of how we marshal values in from YAML, initialize never gets called throughout our codebase. That's what our weird .validate functions do.

At some point, it would be nice to fix this, but that's hardly a priority.